### DISCUSSION TITLE
How do I call /installation/repositories API?

### DISCUSSION THREAD

**Discussion by dougdonohoe** [PyGithub/PyGithub#2104] [Q&A] [answered]
Using PyGithub, is it possible to call the `/installation/repositories` API?

https://docs.github.com/en/rest/reference/apps#installations

**Comment by dougdonohoe** [comment]
So far the only way to do this is to construct an `Installations` object directly:

```
    inst = Installation(
        # Params: login_or_token, password, jwt, base_url, timeout, user_agent, per_page, verify, retry, pool_size
        Requester(token.token, None, None, base_url, 30, "PyGithub/Python", 100, None, None, None), {},
        {}, False)
    rr = inst.get_repos()
    for repo in rr:
        print(f"Repo: {repo.full_name}")
```

**Reply by AleksandraAleksandrova** [reply]
This didn't work for me, but I cannot seem to find anything working.

**Comment by EnricoMi** [comment, marked as answer]
Have you tried:

```python
import github

auth = github.Auth.AppAuth(app_id, private_key)
gi = github.GithubIntegration(auth=auth)
installation = gi.get_app_installation(installation_id)

for repo in installation.get_repos():
  print(repo.full_name)
```

This requires pygithub>=1.59.0.

**Reply by AleksandraAleksandrova** [reply]
this line _auth = github.Auth.AppAuth(app_id, app_key)_ results in AttributeError: module 'github' has no attribute 'Auth . 
When I checked the documentation (https://pygithub.readthedocs.io/en/latest/examples/Authentication.html#app-installation-authentication) is says to use this import _from github import Auth_ , but this raises ImportError: cannot import name 'Auth' from 'github' . I added _import github_ and _from github import *_ to make sure everything is imported and yet this Auth object isn't available.

EDIT:
Solution worked, turns out I had to reinstall my pygithub package, thank you a lot!
